src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -311,6 +288,89 @@ private boolean validateTemporaryAccessToken(String temporaryAccessToken, String
         return true;
     }
 
+    /**
+     * Helper method which handles the file creation for both normal file uploads and for markdown
+     * @param file The file to be uplaoded
+     * @param keepFileName specifies if original file name should be kept
+     * @param markdown boolean which is set to true, when we are uploading a file within the markdown editor
+     * @return The path of the file
+     * @throws URISyntaxException if response path can't be converted into URI
+     */
+    @NotNull
+    private ResponseEntity<String> handleSaveFile(@RequestParam("file") MultipartFile file, @RequestParam("keepFileName") Boolean keepFileName, boolean markdown)
+            throws URISyntaxException {
+        // NOTE: Maximum file size is set in resources/config/application.yml
+        // Currently set to 10 MB
+
+        // check for file type
+        String fileExtension = FilenameUtils.getExtension(file.getOriginalFilename());
+        if (fileExtension == null || this.allowedFileExtensions.stream().noneMatch(fileExtension::equalsIgnoreCase)) {
+            return ResponseEntity.badRequest().body("Unsupported file type! Allowed file types: " + String.join(", ", this.allowedFileExtensions));
+        }
+
+        final String filePath;
+        final String fileNameAddition;
+        final StringBuilder responsePath = new StringBuilder();
+
+        // set the appropriate values depending on the use case
+        if (markdown) {
+            filePath = FilePathService.getMarkdownFilepath();
+            fileNameAddition = "Markdown_";
```

Comment:
How do you handle if two people upload a file with the same name? Does the file name include some random string to prevent that the 2nd file is overridden by the 1st one?

Reply:
The request is for now always sent so that it does not necessarily keep the file name. In this case a timestamp and a random string is inlcuded.

Reply:
what do you mean with "the request is for now always sent"?

Reply:
There is the option to keep the file name. In this case the file with the same name would be overwritten. This is not used at the moment, but it might be used at some point in the future. Right now the name of a markdown file will always include a random string